authentication/auth_method_client_cert.go

```diff
@@ -58,6 +58,8 @@ func (a servicePrincipalClientCertificateAuth) getAuthorizationToken(oauthConfig
 		return nil, err
 	}
 
+	spt.SetSender(sender)
```

Comment:
is it valid to pass `nil`? should you avoid setting it if so?

Reply:
the SDK handles that internally and uses it's own client, as we do today - so this should be fine